ir/instr.cpp

```diff
@@ -227,25 +231,47 @@ StateValue BinOp::toSMT(State &s) const {
     not_poison &= (b == 0u || a != 0u);
     break;
 
+  case SAdd_Overflow:
+    val = a + b;
+    val = val.concat(expr::mkIf(a.add_no_soverflow(b), expr::mkUInt(0, 1), expr::mkUInt(1, 1)));
+    break;
+
+  case ExtractValue:
+    val = expr::mkIf(b == 1u, a.extract(0, 0), a.extract(a.bits() - 1, 1));
```

Comment:
well, this needs to check the type info.

Reply:
we already enforcing the type info in BinOp::getTypeConstraints() for extractvalue. So, we can safely assume `a` to be a aggregate bitvector and `b` to be a BV of size 1, isn't it?

Reply:
okay, this makes sense now after we made our aggregate type generic; i will make it generic here too in my next commit